newIDE/app/src/BehaviorsEditor/Editors/Physics2Editor/index.js

```diff
@@ -215,6 +242,13 @@ const Physics2Editor = (props: Props) => {
               forceUpdate();
             }}
             type="number"
+            endAdornment={
+              <UnitAdornment
+                property={properties.get(
+                  shape === 'Polygon' ? 'PolygonOriginX' : 'shapeDimensionA'

```

Comment:
Something weird in this file.
There is a check at the top of the textfield to display only if `shape !== 'Polygon'` 
And yet we do the check over and over again 🤔

Reply:
Indeed there is a list to chose the origin for polygons.